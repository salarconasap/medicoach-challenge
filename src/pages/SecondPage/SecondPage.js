import React from 'react';
import ScanQR from './components/ScanQR';
import CheckStatus from './components/CheckStatus';
import { useDispatch, useSelector } from 'react-redux';
import { scanQR } from '../../actions/photosActions';
import { CustomLink } from '../../styles'

function SecondPage() {
  const dispatch = useDispatch();
  const { apiserver, api_status } = useSelector( state => state.photos );
  const handleResult = (x) => dispatch(scanQR(x))

  return (
    apiserver ?
      <>
        <CheckStatus server={apiserver}  />
        { api_status && <CustomLink to='/thirdpage'>Take Photo & upload it</CustomLink> } 
      </>
      :
      <ScanQR handleResult={handleResult} />
  );
}

export default SecondPage;
