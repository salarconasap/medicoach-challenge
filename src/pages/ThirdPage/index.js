import ThirdPage from './ThirdPage';

export default ThirdPage