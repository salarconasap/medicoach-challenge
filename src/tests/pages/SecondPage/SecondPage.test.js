import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { SecondPage } from '../../../pages/SecondPage';
// import { apiStatus } from '../../../actions/photosActions';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();

const state = {
    apiserver: null,
    user: null,
    password: null,
    api_status: false,
    img: null
};

const wrapper = () => mount( 
    <Provider store={ store }>
        <SecondPage { ...state }  /> 
    </Provider>
)

describe('test <SecondPage />', () => {
    test('should show correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });
})
