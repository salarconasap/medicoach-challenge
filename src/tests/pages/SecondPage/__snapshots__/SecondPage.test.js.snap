// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <SecondPage /> should show correctly 1`] = `[Function]`;
